lib/src/socket_connector.dart

```diff
@@ -421,27 +468,53 @@ class SocketConnector {
 
     int connections = 0;
     // bind to a local port for side 'A'
-    connector._serverSocketA = await ServerSocket.bind(addressA, portA);
-    // listen on the local port and connect the inbound socket
-    connector._serverSocketA?.listen((sideASocket) async {
-      if (!multi) {
-        unawaited(connector._serverSocketA?.close());
-      }
+    connector._serverSocketA = await ServerSocket.bind(
+      addressA,
+      portA,
+      backlog: backlog,
+    );
+
+    StreamController<Socket> ssc = StreamController();
+    ssc.stream.listen((sideASocket) async {
```

Comment:
The code in this listen closure is identical to what was previously in the serverSocketA.listen closure